react/src/lib/components/DeckGLMap/components/Map.tsx

```diff
@@ -323,19 +323,27 @@ const Map: React.FC<MapProps> = ({
     const deckRef = useRef<DeckGLRef>(null);
     const bboxInitial: BoundingBox = [0, 0, 0, 1, 1, 1];
     const boundsInitial = bounds ?? [0, 0, 1, 1];
-
     // state for views prop of DeckGL component
     const [viewsProps, setViewsProps] = useState<ViewportType[]>([]);
     const [alteredLayers, setAlteredLayers] = useState([{}]);
 
+    const isViewsDefined = checkIfViewsIsDefined();
     const initialViewState = getViewState(
         boundsInitial,
-        views?.viewports[0].target,
-        views?.viewports[0].zoom,
+        isViewsDefined ? views?.viewports[0].target : undefined,
+        isViewsDefined ? views?.viewports[0].zoom : undefined,
         deckRef.current?.deck
     );
 
     // Local help function.
+    function checkIfViewsIsDefined() {
```

Comment:
Please avoid local functions in render loop, because it can affect render performance. This function can be module scoped if parameterized.

Reply:
I have move this function outside the React.FC. in the meantime I am thinking if we should also remove the calcDefaultViewStates function since it is also a local helper function ? Correct me if I am understand it in the wrong way....